carbon-design-system/carbon-addons-iot-react
# fix(hierarchylist): search does not match catagories

Closes #1421

**Summary**
Fix to add the category to the search result. 

Functional spec as I interpret the request:
If the user search term matches a category content value, that category and all child items of that category should be added to the list of filtered search result. If the matching category also have child items that matches the search term they should not appear a second time in the list.

**Change List (commits, features, bugs, etc)**
- Modified the HierarchyList
- Added unit tests

**Acceptance Test (how to verify the PR)**
1. Go to story [watson-iot-experimental-hierarchylist--stateful-list-with-nested-searching](https://deploy-preview-1665--carbon-addons-iot-react.netlify.app?path=/story/watson-iot-experimental-hierarchylist--stateful-list-with-nested-searching)
2. Type "ne" in the search field and make sure the categories "New York Yankees" and "New York Mets" are shown together with all child elements. 
3. Make sure the categories only appear once, and make sure that "Chicago White Sox" and "Washington Nationals" are also listed with players matching the search.
3. Type "new"  in the search field and make sure the categories "New York Yankees" and "New York Mets" are shown together with all child elements, but no other categories are shown.
